InfiniteLoopLMSSQL/src/com/st/lmssql/service/BorrowerService.java

```diff
@@ -0,0 +1,225 @@
+	
+	
+	public boolean borCardNoExists(int cardNo) {
+		boolean flag;
+		Borrower bor = null;
+		try {
+			bor = genDaoBorrower.get(cardNo);
+			con.commit();
+		} catch (SQLException e) {
+			System.err.println("Problem with checking card number!");
+			myRollBack();
+		}
+		
+		if(bor.getName() == null) 
+			flag = false;
+		else 
+			flag = true;
+		
+		return flag;
+	}
+	
+	public List<LibraryBranch> getAllBranches() {
+		List<LibraryBranch> libBranches = null;
+		try {
+			libBranches = genDaoLibBranch.getAll();
+			con.commit();
+		} catch (SQLException e) {
+			System.out.println("Get All Branches Failed! :(");
+			myRollBack();
+		}
+		return libBranches;
+	}
+	
+	//returns all bookCopies>=1 specific to a branch with book and author
+	public List<BkCopiesDTO> getBkCopiesGreater1BookAndTitle(int branchId) {
+		List<BkCopiesDTO> list = null;
+		try {
+			List<BookCopies> bookCopies = bookCopiesDao.getAll();
+			Book book;
+			Author author;
+			
+			BkCopiesDTO obj;
+			list= new ArrayList<>();
+			
+			for(BookCopies bc : bookCopies) {			
+				if(bc.getBranchId()==branchId && bc.getNoOfCopies()>=1) {
+					//kind of like doing joins
+					book = genDaoBook.get(bc.getBookId());
+					author = genDaoAuthor.get(book.getAuthorId());
+					
+					obj = new BkCopiesDTO(bc, book, author);
+					list.add(obj);
+				}
+			}
+			con.commit();
+		} 
+		catch(SQLException e) {
+			System.err.println("Unable to load books from the branch!");
+			myRollBack();
+		}
+		
+		return list;
+	}
+	
+	public void checkOutBook(int bookId, int branchId, int cardNo, Date dateOut, Date dueDate, int noOfCopies) {
+		BookLoans bl = new BookLoans(bookId, branchId, cardNo, dateOut, dueDate);
+		BookCopies bc = new BookCopies(bookId, branchId, noOfCopies);
+		try {
+			bookLoansDao.add(bl);      //add an entry to book_loans
+			bookCopiesDao.update(bc);  //update noOfCopies with 1 less of that book
+			con.commit();
+			System.out.println("Book Checked Out!");
+		} catch (SQLException e) {
+			System.out.println("Unable to check out book!");
+			myRollBack();
+		}
+	}
+	
+	/*##############################################################*/
+	
+	//returns the branches that user has a book checked out from based on their cardNo
+	public List<BkLoansBranchDTO> getBranchesWithBkLoans(int cardNo) {
+		List<BkLoansBranchDTO> list = null;
+		try {
+			List<BookLoans> bookLoans = bookLoansDao.getAll();
+			LibraryBranch libBranch;
+			
+			BkLoansBranchDTO obj;
+			list = new ArrayList<>();
+			
+			for(BookLoans bl : bookLoans) {
+				libBranch = genDaoLibBranch.get(bl.getBranchId());
+				
+				if(bl.getCardNo()==cardNo && bl.getDateOut()!=null && bl.getDueDate()!=null) {
+					obj = new BkLoansBranchDTO(bl, libBranch);
+					list.add(obj);
+				}
+			}
+			
+			//remove duplicates based on branch name
+			HashSet<String> seen = new HashSet<>();
+			list.removeIf(e -> !seen.add(e.getLibBranch().getBranchName()));
+			
+			con.commit();
+		} 
+		catch(SQLException e) {

```

Comment:
You may want to consider doing what Chris mentioned and catching these SQLExceptions and throwing your own custom exceptions to handle in menu.